### PR TITLE
Story integration

### PR DESCRIPTION
**Pull request by ritz078**
implemented photostory as a higher order component.

### CODE REVIEW

**Comment by ritz078** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://fb5a67cf-39d1-4e55-9a24-57a4ec689004.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/housinghq/react-photostory/branch/story/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: components/Story.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import autoBind from 'react-auto-bind'
+
+export default class Story extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isOpen: false,
+      currentIndex: 0,
+      length: 0,
+      pan: null,
```
Comment (pranjal-jain) [inline components/Story.js]: Change pan to panImgUrl

File: components/Story.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    autoBind(this)
+  }
+
+  handleClick ({index, length, title, subTitle, image, data}) {
+    const {isOpen} = this.state
+    if (!isOpen) {
+      this.setState({isOpen: true, length, currentIndex: index, title, subTitle, data}, () => {
+        document.body.style.overflow = this.state.isOpen ? 'hidden' : null
```
Comment (pranjal-jain) [inline components/Story.js:25]: in componentWillUnmount add document.body.style.overflow = null

File: components/Story.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+export default class Story extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isOpen: false,
+      currentIndex: 0,
+      length: 0,
+      pan: null,
+      data: {}
+    }
```
Comment (pranjal-jain) [inline components/Story.js:13]: data, length, title, subtitle can be moved out of state

File: components/Story.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  handleClick ({index, length, title, subTitle, image, data}) {
+    const {isOpen} = this.state
+    if (!isOpen) {
+      this.setState({isOpen: true, length, currentIndex: index, title, subTitle, data}, () => {
+        document.body.style.overflow = this.state.isOpen ? 'hidden' : null
+      })
+
+      this.props.onOpen({index, title, subTitle, data})
```
Comment (pranjal-jain) [inline components/Story.js:28]: Check for undefined before calling

File: components/Story.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+  disablePan () {
+    this.setState({pan: false})
+  }
+
+  closeModal () {
+    this.setState({
+      isOpen: false
+    }, () => {
+      this.props.onClose({index: this.state.currentIndex})
```
Comment (pranjal-jain) [inline components/Story.js:49]: Check for undefined before calling

File: components/Swipe.js [hunk 6]
```diff
@@ -151,8 +155,17 @@ export default class Swipe extends Component {
   }
 
   handleClick () {
+    setTimeout(this.setWidth, 0)
```
Comment (pranjal-jain) [inline components/Swipe.js:158]: setWidth can be called in onOpen callback

**Review by pranjal-jain — CHANGES_REQUESTED** [review]

File: components/Story.js (hunk 4 above)
Comment (ritz078) [reply components/Story.js:28]: not needed. empty function is present in defaultProps

File: components/Story.js (hunk 5 above)
Comment (ritz078) [reply components/Story.js:49]: same

**Pull request merged**